contracts/truefi2/PoolFactory.sol

```diff
@@ -0,0 +1,92 @@
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {IImplementationReference} from "../proxy/interface/IImplementationReference.sol";
+
+contract PoolFactory is Ownable {
+    // ================ WARNING ==================
+    // ===== THIS CONTRACT IS INITIALIZABLE ======
+    // === STORAGE VARIABLES ARE DECLARED BELOW ==
+    // REMOVAL OR REORDER OF VARIABLES WILL RESULT
+    // ========= IN STORAGE CORRUPTION ===========
+
+    // @dev Mapping of ERC20 token's addresses to its pool's addresses
+    mapping(address => address) public correspondingPool;
+    mapping(address => bool) public isPool;
+
+    // @dev Whitelist for tokens, which can have pools created
+    mapping(address => bool) public isAllowed;
+
+    IImplementationReference public poolImplementationReference;
+
+    // ======= STORAGE DECLARATION END ===========
+
+    /**
+     * @dev Event to show creation of the new pool
+     * @param token Address of token, for which the pool was created
+     * @param pool Address of new pool's proxy
+     */
+    event PoolCreated(address token, address pool);
+
+    /**
+     * @dev Event to show that token is now allowed/disallowed to have a pool created
+     * @param token Address of token
+     * @param status New status of allowance
+     */
+    event AllowedStatusChanged(address token, bool status);
+
+    /**
+     * @dev Throws if token already has an existing corresponding pool
+     * @param token Token to be checked for existing pool
+     */
+    modifier onlyNotExistingPools(address token) {
+        require(correspondingPool[token] == address(0), "PoolFactory: This token already has a corresponding pool");
+        _;
+    }
+
+    modifier onlyAllowed(address token) {
+        require(isAllowed[token] == true, "PoolFactory: This token is not allowed to have a pool");
+        _;
+    }
+
+    /**
+     * @dev Initialize this contract with provided parameters
+     * @param _poolImplementationReference First implementation reference of TrueFiPool
+     */
+    function initialize(IImplementationReference _poolImplementationReference) external initializer {
+        Ownable.initialize();
+
+        poolImplementationReference = _poolImplementationReference;
+    }
+
+    /**
+     * @dev Create a new pool behind proxy. Update new pool's implementation.
+     * Transfer ownership of created pool to Factory owner.
+     * @param token Address of token which the pool will correspond to.
+     */
+    function createPool(address token) external onlyAllowed(token) onlyNotExistingPools(token) {
+        OwnedProxyWithReference proxy = new OwnedProxyWithReference();
+        proxy.changeImplementationReference(poolImplementationReference);
+        ITrueFiPool2(address(proxy)).initialize(ERC20(token), msg.sender);
+        proxy.transferProxyOwnership(this.owner());
```

Comment:
This ownership transfer only sets pending owner. Maybe we should add owner setter to proxy constructor? And implementation for good measure